Cannot open module file ‘dftd4.mod’ when building cp2k

CP2K version: `v2025.2`

Hi!

I'm trying to build CP2K with DFTD4, and I have encountered some compilation errors. I'm not sure whether it's a bug or I have done something wrong. Here is what I did:

Build toolchain:
```bash
(base) liuheqi@liuheqi-lxd:~/softwares/cp2k-2025.2/tools/toolchain$ ./install_cp2k_toolchain.sh \
-j 100 \
--enable-cuda --gpu-ver=A100 \
--with-openmpi=install \
--with-ninja=install \
--with-plumed=install \
--with-dftd4=install
```

Build toolchain Output:

```
WARNING: (./install_cp2k_toolchain.sh, line 371) No MPI installation detected (ignore this message in Cray Linux Environment or when MPI installation was requested).
Compiling with 100 processes for target native.
==================== Finding GCC from system paths ====================
path to gcc is /usr/bin/gcc
path to g++ is /usr/bin/g++
path to gfortran is /usr/bin/gfortran
GCC compiler version 13.3.0 found
Found include directory /usr/include
Step gcc took 0.00 seconds.
Step intel took 0.00 seconds.
Step amd took 0.00 seconds.
==================== Getting proc arch info using OpenBLAS tools ====================
wget  --quiet https://www.cp2k.org/static/downloads/OpenBLAS-0.3.30.tar.gz -O OpenBLAS-0.3.30.tar.gz
OpenBLAS-0.3.30.tar.gz: OK
Checksum of OpenBLAS-0.3.30.tar.gz Ok
OpenBLAS detected LIBCORE = zen
OpenBLAS detected ARCH    = x86_64
==================== Installing CMake ====================
wget  --quiet https://www.cp2k.org/static/downloads/cmake-3.31.7-linux-x86_64.sh -O cmake-3.31.7-linux-x86_64.sh
cmake-3.31.7-linux-x86_64.sh: OK
Checksum of cmake-3.31.7-linux-x86_64.sh Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/cmake-3.31.7
Step cmake took 7.00 seconds.
==================== Installing Ninja  ====================
wget  --quiet https://www.cp2k.org/static/downloads/ninja-v1.13.1.tar.gz -O ninja-v1.13.1.tar.gz
ninja-v1.13.1.tar.gz: OK
Checksum of ninja-v1.13.1.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/ninja-v1.13.1
Step ninja took 9.00 seconds.
==================== Installing OpenMPI ====================
wget  --quiet https://www.cp2k.org/static/downloads/openmpi-5.0.8.tar.bz2 -O openmpi-5.0.8.tar.bz2
openmpi-5.0.8.tar.bz2: OK
Checksum of openmpi-5.0.8.tar.bz2 Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8
Found directory /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin
Found directory /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/lib
Found directory /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/include
mpiexec is installed as /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpiexec
mpicc is installed as /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpicc
mpicxx is installed as /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpicxx
mpifort is installed as /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpifort
Step openmpi took 489.00 seconds.
==================== Installing OpenBLAS ====================
OpenBLAS-0.3.30.tar.gz is found
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openblas-0.3.30
Installing OpenBLAS library for target ZEN
Step openblas took 132.00 seconds.
Step gmp took 0.00 seconds.
==================== Installing FFTW ====================
wget  --quiet https://www.cp2k.org/static/downloads/fftw-3.3.10.tar.gz -O fftw-3.3.10.tar.gz
fftw-3.3.10.tar.gz: OK
Checksum of fftw-3.3.10.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/fftw-3.3.10
Step fftw took 63.00 seconds.
==================== Installing LIBINT ====================
wget  --quiet https://www.cp2k.org/static/downloads/libint-v2.6.0-cp2k-lmax-5.tgz -O libint-v2.6.0-cp2k-lmax-5.tgz
libint-v2.6.0-cp2k-lmax-5.tgz: OK
Checksum of libint-v2.6.0-cp2k-lmax-5.tgz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libint-v2.6.0-cp2k-lmax-5
Step libint took 186.00 seconds.
==================== Installing LIBXC ====================
wget  --quiet https://www.cp2k.org/static/downloads/libxc-7.0.0.tar.bz2 -O libxc-7.0.0.tar.bz2
libxc-7.0.0.tar.bz2: OK
Checksum of libxc-7.0.0.tar.bz2 Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxc-7.0.0
Step libxc took 1315.00 seconds.
Step greenx took 0.00 seconds.
==================== Using libgrpp ====================
Step libgrpp took 0.00 seconds.
==================== Installing Libxsmm ====================
wget  --quiet https://www.cp2k.org/static/downloads/libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec.tar.gz -O libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec.tar.gz
libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec.tar.gz: OK
Checksum of libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec
Step libxsmm took 16.00 seconds.
==================== Installing ScaLAPACK ====================
wget  --quiet https://www.cp2k.org/static/downloads/scalapack-2.2.2.tar.gz -O scalapack-2.2.2.tar.gz
scalapack-2.2.2.tar.gz: OK
Checksum of scalapack-2.2.2.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/scalapack-2.2.2
Step scalapack took 14.00 seconds.
==================== Installing COSMA ====================
wget  --quiet https://www.cp2k.org/static/downloads/COSMA-v2.7.0.tar.gz -O COSMA-v2.7.0.tar.gz
COSMA-v2.7.0.tar.gz: OK
Checksum of COSMA-v2.7.0.tar.gz Ok
wget  --quiet https://www.cp2k.org/static/downloads/COSTA-v2.2.2.tar.gz -O COSTA-v2.2.2.tar.gz
COSTA-v2.2.2.tar.gz: OK
Checksum of COSTA-v2.2.2.tar.gz Ok
wget  --quiet https://www.cp2k.org/static/downloads/Tiled-MM-v2.3.1.tar.gz -O Tiled-MM-v2.3.1.tar.gz
Tiled-MM-v2.3.1.tar.gz: OK
Checksum of Tiled-MM-v2.3.1.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/COSMA-2.7.0
Step cosma took 47.00 seconds.
==================== Installing ELPA ====================
wget  --quiet https://www.cp2k.org/static/downloads/elpa-2024.05.001.tar.gz -O elpa-2024.05.001.tar.gz
elpa-2024.05.001.tar.gz: OK
Checksum of elpa-2024.05.001.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/cpu
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia
Step elpa took 367.00 seconds.
==================== Installing GSL ====================
wget  --quiet https://www.cp2k.org/static/downloads/gsl-2.8.tar.gz -O gsl-2.8.tar.gz
gsl-2.8.tar.gz: OK
Checksum of gsl-2.8.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/gsl-2.8
Step gsl took 64.00 seconds.
==================== Installing PLUMED ====================
wget  --quiet https://www.cp2k.org/static/downloads/plumed-src-2.9.3.tgz -O plumed-src-2.9.3.tgz
plumed-src-2.9.3.tgz: OK
Checksum of plumed-src-2.9.3.tgz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/plumed-2.9.3
Step plumed took 70.00 seconds.
Step libtorch took 0.00 seconds.
Step deepmd took 0.00 seconds.
Step ace took 0.00 seconds.
==================== Installing hdf5 ====================
wget  --quiet https://www.cp2k.org/static/downloads/hdf5-1.14.6.tar.gz -O hdf5-1.14.6.tar.gz
hdf5-1.14.6.tar.gz: OK
Checksum of hdf5-1.14.6.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/hdf5-1.14.6
Step hdf5 took 68.00 seconds.
==================== Installing libvdwxc ====================
wget  --quiet https://www.cp2k.org/static/downloads/libvdwxc-0.4.0.tar.gz -O libvdwxc-0.4.0.tar.gz
libvdwxc-0.4.0.tar.gz: OK
Checksum of libvdwxc-0.4.0.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvdwxc-0.4.0
Step libvdwxc took 25.00 seconds.
==================== Installing Spglib ====================
wget  --quiet https://www.cp2k.org/static/downloads/spglib-2.5.0.tar.gz -O spglib-2.5.0.tar.gz
spglib-2.5.0.tar.gz: OK
Checksum of spglib-2.5.0.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/spglib-2.5.0
Step spglib took 5.00 seconds.
==================== Installing libvori ====================
wget  --quiet https://www.cp2k.org/static/downloads/libvori-220621.tar.gz -O libvori-220621.tar.gz
libvori-220621.tar.gz: OK
Checksum of libvori-220621.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvori-220621
Step libvori took 17.00 seconds.
Step libsmeagol took 0.00 seconds.
==================== Installing pugixml ====================
wget  --quiet https://www.cp2k.org/static/downloads/pugixml-1.15.tar.gz -O pugixml-1.15.tar.gz
pugixml-1.15.tar.gz: OK
Checksum of pugixml-1.15.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/pugixml-1.15
Step pugixml took 8.00 seconds.
==================== Installing SpFFT ====================
wget  --quiet https://www.cp2k.org/static/downloads/SpFFT-1.1.1.tar.gz -O SpFFT-1.1.1.tar.gz
SpFFT-1.1.1.tar.gz: OK
Checksum of SpFFT-1.1.1.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpFFT-1.1.1
Step spfft took 15.00 seconds.
==================== Installing SpLA ====================
wget  --quiet https://www.cp2k.org/static/downloads/SpLA-1.6.1.tar.gz -O SpLA-1.6.1.tar.gz
SpLA-1.6.1.tar.gz: OK
Checksum of SpLA-1.6.1.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpLA-1.6.1
Step spla took 17.00 seconds.
==================== Installing SIRIUS ====================
wget  --quiet https://www.cp2k.org/static/downloads/SIRIUS-7.7.0.tar.gz -O SIRIUS-7.7.0.tar.gz
SIRIUS-7.7.0.tar.gz: OK
Checksum of SIRIUS-7.7.0.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/sirius-7.7.0
patching file src/core/memory.hpp
Step sirius took 394.00 seconds.
==================== Installing DFTD4 ====================
wget  --quiet https://www.cp2k.org/static/downloads/dftd4-3.7.0.tar.gz -O dftd4-3.7.0.tar.gz
dftd4-3.7.0.tar.gz: OK
Checksum of dftd4-3.7.0.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dftd4-3.7.0
Step dftd4 took 15.00 seconds.
Step trexio took 0.00 seconds.
Step tblite took 0.00 seconds.
==================== Installing DBCSR ====================
wget  --quiet https://github.com/cp2k/dbcsr/releases/download/v2.8.0/dbcsr-2.8.0.tar.gz -O dbcsr-2.8.0.tar.gz
dbcsr-2.8.0.tar.gz: OK
Checksum of dbcsr-2.8.0.tar.gz Ok
Installing from scratch into /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dbcsr-2.8.0
Step DBCSR took 111.00 seconds.
==================== generating arch files ====================
arch files can be found in the /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch subdirectory
path to nvcc is /usr/local/cuda-12.8/bin/nvcc
libcudart is found in ld search path
libnvrtc is found in ld search path
libcuda is found in ld search path
libcufft is found in ld search path
libcublasLt is found in ld search path
libcublas is found in ld search path
Found lib directory /usr/local/cuda-12.8/targets/x86_64-linux/lib
Found lib directory /usr/local/cuda-12.8/targets/x86_64-linux/lib
Found lib directory /usr/local/cuda-12.8/targets/x86_64-linux/lib
Found lib directory /usr/local/cuda-12.8/targets/x86_64-linux/lib
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local.ssmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_static.ssmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local.sdbg
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_asan.ssmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_coverage.sdbg
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local.psmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local.pdbg
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_asan.psmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_static.psmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_warn.psmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_coverage.pdbg
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_cuda.ssmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_cuda.sdbg
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_cuda.psmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_cuda.pdbg
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_cuda_warn.psmp
Wrote /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/local_coverage_cuda.pdbg
========================== usage =========================
Done!
Now copy:
  cp /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/arch/* to the cp2k/arch/ directory
To use the installed tools and libraries and cp2k version
compiled with it you will first need to execute at the prompt:
  source /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/setup
To build CP2K you should change directory:
  cd cp2k/
  make -j 100 ARCH=local VERSION="ssmp sdbg psmp pdbg"

arch files for GPU enabled CUDA versions are named "local_cuda.*"
arch files for GPU enabled HIP versions are named "local_hip.*"
arch files for OpenCL (GPU) versions are named "local_opencl.*"
arch files for coverage versions are named "local_coverage.*"

Note that these pre-built arch files are for the GNU compiler, users have to adapt them for other compilers.
It is possible to use the provided CP2K arch files as guidance.
```

Command2:
```bash
(base) liuheqi@liuheqi-lxd:~/softwares/cp2k-2025.2$ cp tools/toolchain/install/arch/* arch/
(base) liuheqi@liuheqi-lxd:~/softwares/cp2k-2025.2$ source /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/setup
(base) liuheqi@liuheqi-lxd:~/softwares/cp2k-2025.2$ make -j 100 ARCH=local_cuda VERSION=psmp
```

And I met an compilation error:

```
Fatal Error: Cannot open module file ‘dftd4.mod’ for reading at (1): No such file or directory
compilation terminated.
make[3]: *** [/home/liuheqi/softwares/cp2k-2025.2/Makefile:516: qs_dispersion_d4.o] Error 1
```

But I can find `dftd4.mod`:

```bash
(base) liuheqi@liuheqi-lxd:~/softwares/cp2k-2025.2$ find . -name "dftd4.mod"
./tools/toolchain/build/dftd4-3.7.0/subprojects/dftd4/build/include/dftd4.mod
./tools/toolchain/install/dftd4-3.7.0/include/dftd4/GNU-13.3.0/dftd4.mod
```

This is the `arch/local_cuda.psmp`:

```
CC          = /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpicc
CXX         = /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpicxx
AR          = ar -r
FC          = /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpifort
LD          = /home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/bin/mpifort
#
DFLAGS      = -D__OFFLOAD_CUDA -D__DBCSR_ACC  -D__LIBXSMM  -D__parallel -D__MPI_F08 -D__FFTW3  -D__LIBINT -D__LIBXC -D__LIBGRPP -D__parallel -D__COSMA -D__ELPA -D__ELPA_NVIDIA_GPU -D__GSL -D__PLUMED2 -D__HDF5 -D__LIBVDWXC -D__SPGLIB -D__LIBVORI -D__SPFFT  -D__OFFLOAD_GEMM  -D__SPLA -D__SIRIUS -D__DFTD4 -D__DBCSR_ACC -D__DBCSR  
#
WFLAGS      = -Werror=aliasing -Werror=ampersand -Werror=c-binding-type -Werror=intrinsic-shadow -Werror=intrinsics-std -Werror=line-truncation -Werror=tabs -Werror=target-lifetime -Werror=underflow -Werror=unused-but-set-variable -Werror=unused-variable -Werror=unused-dummy-argument -Werror=unused-parameter -Werror=unused-label -Werror=conversion -Werror=zerotrip -Wno-maybe-uninitialized -Wuninitialized -Wuse-without-only 
#
FCDEBFLAGS  = -fbacktrace -ffree-form -fimplicit-none -std=f2008 
CFLAGS      = -fno-omit-frame-pointer -fopenmp -g -march=native -mtune=native  -O3 -funroll-loops $(PROFOPT)  -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/include'  -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openblas-0.3.30/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/fftw-3.3.10/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libint-v2.6.0-cp2k-lmax-5/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxc-7.0.0/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/COSMA-2.7.0/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia/include/elpa_openmp-2024.05.001/modules' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia/include/elpa_openmp-2024.05.001/elpa' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/gsl-2.8/include' -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/hdf5-1.14.6/include -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvdwxc-0.4.0/include -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/spglib-2.5.0/include -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/pugixml-1.15/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpFFT-1.1.1/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpLA-1.6.1/include/spla' -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/sirius-7.7.0/cuda/include/sirius -I'' -I'' -I'' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dbcsr-2.8.0/include' -std=c17 -Wall -Wextra -Wno-vla-parameter -Wno-deprecated-declarations $(DFLAGS) -I/usr/local/cuda-12.8/include
CXXFLAGS    = -O2 -fPIC -fno-omit-frame-pointer -fopenmp -g -mtune=native -std=c++17 $(DFLAGS) -Wno-deprecated-declarations -I/usr/local/cuda-12.8/include
FCFLAGS     = -fno-omit-frame-pointer -fopenmp -g -march=native -mtune=native  -O3 -funroll-loops $(PROFOPT)  -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/include'  -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openblas-0.3.30/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/fftw-3.3.10/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libint-v2.6.0-cp2k-lmax-5/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxc-7.0.0/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/COSMA-2.7.0/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia/include/elpa_openmp-2024.05.001/modules' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia/include/elpa_openmp-2024.05.001/elpa' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/gsl-2.8/include' -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/hdf5-1.14.6/include -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvdwxc-0.4.0/include -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/spglib-2.5.0/include -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/pugixml-1.15/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpFFT-1.1.1/include' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpLA-1.6.1/include/spla' -I/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/sirius-7.7.0/cuda/include/sirius -I'' -I'' -I'' -I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dbcsr-2.8.0/include' $(FCDEBFLAGS) $(WFLAGS) $(DFLAGS) -fallow-argument-mismatch
#
LDFLAGS     =  $(FCFLAGS) -Wl,--enable-new-dtags -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openmpi-5.0.8/lib'  -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openblas-0.3.30/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/openblas-0.3.30/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/fftw-3.3.10/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/fftw-3.3.10/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libint-v2.6.0-cp2k-lmax-5/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxc-7.0.0/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxc-7.0.0/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libxsmm-e0c4a2389afba36c453233ad7de07bd92c715bec/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/scalapack-2.2.2/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/scalapack-2.2.2/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/COSMA-2.7.0-cuda/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/COSMA-2.7.0-cuda/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/elpa-2024.05.001/nvidia/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/gsl-2.8/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/gsl-2.8/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/plumed-2.9.3/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/plumed-2.9.3/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/hdf5-1.14.6/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/hdf5-1.14.6/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvdwxc-0.4.0/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvdwxc-0.4.0/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/spglib-2.5.0/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/spglib-2.5.0/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvori-220621/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/libvori-220621/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/pugixml-1.15/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/pugixml-1.15/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpFFT-1.1.1/lib/cuda' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpFFT-1.1.1/lib/cuda' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpLA-1.6.1/lib/cuda' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/SpLA-1.6.1/lib/cuda' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/sirius-7.7.0/cuda/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/sirius-7.7.0/cuda/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dftd4-3.7.0/lib' -Wl,-rpath,'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dftd4-3.7.0/lib' -L'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dbcsr-2.8.0/lib' -Wl,-rpath='/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dbcsr-2.8.0/lib' -L'/usr/local/cuda-12.8/targets/x86_64-linux/lib' -Wl,-rpath,'/usr/local/cuda-12.8/targets/x86_64-linux/lib'
LDFLAGS_C   = 
LIBS        = -ldbcsr -ldftd4 -lmstore -lmulticharge -lmctc-lib -lsirius -lsirius_cxx -lcusolver -lspla -lspfft -lpugixml -lsymspg -lvdwxc -l:libhdf5_fortran.a -l:libhdf5_f90cstub.a -l:libhdf5.a -lz -l:libplumed.a -ldl -lstdc++ -lz -ldl -lgsl -lelpa_openmp -lcosma_prefixed_pxgemm -lcosma -lcosta -lTiled-MM -lscalapack -lxsmmf -lxsmmext -lxsmm -ldl -lpthread -lxcf03 -lxc -lint2 -lfftw3_mpi -lfftw3 -lfftw3_omp   -lmpi  -l:libopenblas.a -lvori -lstdc++ -lstdc++ -lcudart -lnvrtc -lcuda -lcufft -lcublasLt -lcublas -lrt 
#
GPUVER        = A100
OFFLOAD_CC    = nvcc
OFFLOAD_FLAGS = -g -arch sm_80 -O3 -allow-unsupported-compiler -Xcompiler='-fopenmp -Wall -Wextra' --std=c++11 $(DFLAGS) 
OFFLOAD_TARGET = cuda
#
FYPPFLAGS   = -n --line-marker-format=gfortran5
```

So I add:

```
-I'/home/liuheqi/softwares/cp2k-2025.2/tools/toolchain/install/dftd4-3.7.0/include/dftd4/GNU-13.3.0'
```

in  `CFLAGS =` and `FCFLAGS =`, and successfully build.

I'm not familiar with Fortran's build system, so I'm not sure if this is the correct way to do it, but it did get it to compile. I'm currently running tests to see if everything works properly. If this is indeed a bug, I hope we can work together to fix this. 😺

--- a/tools/toolchain/scripts/stage8/install_dftd4.sh
+++ b/tools/toolchain/scripts/stage8/install_dftd4.sh
@@ -96,13 +96,13 @@ EOF
 
   if [ "$with_dftd4" != "__SYSTEM__" ]; then
     TEMP_LOC=$(find ${pkg_install_dir}/include -name "multicharge.mod")
-    MCHARGE=${DFTD4_LOC%/*}
+    MCHARGE=${TEMP_LOC%/*}
     TEMP_LOC=$(find ${pkg_install_dir}/include -name "mstore.mod")
-    MSTORE=${DFTD4_LOC%/*}
+    MSTORE=${TEMP_LOC%/*}
     TEMP_LOC=$(find ${pkg_install_dir}/include -name "mctc_io.mod")
-    MCTC=${DFTD4_LOC%/*}
+    MCTC=${TEMP_LOC%/*}
     TEMP_LOC=$(find ${pkg_install_dir}/include -name "dftd4.mod")
-    DFTD4=${DFTD4_LOC%/*}
+    DFTD4=${TEMP_LOC%/*}
 
     DFTD4_INCLUDE_DIRS="$pkg_install_dir/include"
     DFTD4_LINK_LIBRARIES="${pkg_install_dir}/lib"

